### DISCUSSION TITLE
Trying to set sync\watch but too dumb :)

### DISCUSSION THREAD

**Discussion by KureOnGit** [Taxel/PlexTraktSync#1769] [Q&A] [answered]
I used to run the sync through a cron job but re-installed through pipx for a cleaner install/easier to update and now I'm not sure how to call it. (I've tried just plextraktsync and plextraktsync sync, no dice, cron needs the fulll path I guess).
I checked the info output and the default folders for pipx but can't find the .sh location.

I then searched around here in this discussion and realized watch could be used instead (although it doesn't seem to sync list and stuff) so I tried to create a service but keep getting "Failed to start PlexTraktSync.service: Unit PlexTraktSync.service has a bad unit file setting."
For "ExecStart" I have plextraktsync watch, not sure if this is supported with a pipx instalation?
I would guess so since that is what I used manually, so maybe the error is something else? My searches on this end in ppl having problems with ExecStart such as not complete path or unsupported empty spaces.

I guess ideally I'd get the watch running and sync every month or so instead of the 2h interval I had before.
If anyone has some pointers I'd appreciate it.

**Comment by glensc** [comment]
instead of long essay what you had for breakfast, show the details of the files and actual errors and other relevant information. details matter.

**Comment by KureOnGit** [comment, marked as answer]
So if anyone finds this thread wondering where the installation for pipx is: /home/pi/.local/pipx/venvs/plextraktsync/bin/plextraktsync

I had to use this path for both the systemd service and the cronjob.

**Reply by glensc** [reply]
no. you got it wrong. you haven't completed pipx installation correctly.

pipx puts wrapper to a location that's supposed to be in your PATH

```
➜ pipx list
venvs are in /Users/glen/.local/pipx/venvs
apps are exposed on your $PATH at /Users/glen/.local/bin
manual pages are exposed at /Users/glen/.local/share/man
   package plextraktsync 0.28.12, installed using Python 3.12.1
    - plextraktsync
```

```
➜ which plextraktsync
/Users/glen/.local/bin/plextraktsync
```
    
it's all documented in pipx readme:
- https://github.com/pypa/pipx#install-pipx

```
➜ pipx ensurepath
/Users/glen/.local/bin is already in PATH.

⚠️  All pipx binary directories have been added to PATH. If you are sure you want to proceed, try again with the '--force' flag.

Otherwise pipx is ready to go! ✨ 🌟 ✨
```

**Comment by KureOnGit** [comment]
Yep, run the same commands and confirms the paths I said.
Maybe because I'm stuck with python 3.7 so perhaps my answer is only valid for someone running raspOS x32

**Reply by glensc** [reply]
I assume you replied to my reply:
- https://github.com/Taxel/PlexTraktSync/discussions/1769#discussioncomment-8331780

why you started another thread?

what I tried to say you are using internal path of pipx venv, but should use the wrapper it created.

in my example:
- wrapper: /Users/glen/.local/bin/plextraktsync
- internal path: /Users/glen/.local/pipx/venvs/plextraktsync/bin/plextraktsync

and you should use the wrapper path, not internal path.

what does your `list` and `ensurepath` show?

**Reply by KureOnGit** [reply]
Didn't notice there a specific place to reply.
Ye you're right, there is a shortcut at /.local/bin.
Doesn't come up when doing a sudo find -f

**Reply by glensc** [reply]
you should correct your answer then. either edit it or choose different reply as answer.

**Reply by glensc** [reply]
what is sudo find -f? you mean find -type f? surely it won't find it because it's a symlink

```
$ ls -l /home/glen/.local/bin/plextraktsync
lrwxrwxrwx 1 glen glen 45 Nov  4 20:56 /home/glen/.local/bin/plextraktsync -> ../pipx/venvs/plextraktsync/bin/plextraktsync
```

**Reply by glensc** [reply]
and didn't pipx install tell you location where it placed the file? why you needed to resort to find? and still missing output of your pipx ensurepath
